cpp/src/spatial/hausdorff.cu

```diff
@@ -103,49 +135,92 @@ struct hausdorff_functor {
+
+    // ===== Make Hausdorff Accumulator ============================================================
 
-    // utilize one block per result (pair of spaces).
-    size_type num_blocks_x = min(result->size(), MAX_NUM_BLOCKS_X);
-    size_type num_blocks_y = ceil(result->size() / (float)MAX_NUM_BLOCKS_X);
+    auto num_distances = static_cast<int32_t>(num_points) * static_cast<int32_t>(num_points);
 
-    dim3 grid(num_blocks_x, num_blocks_y);
+    auto hausdorff_index_iter = thrust::make_transform_iterator(
+      thrust::make_counting_iterator<int32_t>(0),
+      haus_traversal_functor<decltype(space_offset_iterator)>{num_spaces,
+                                                              num_points,
+                                                              space_offsets.data<size_type>(),
+                                                              temp_space_lookup.data().get(),
+                                                              space_offset_iterator});
 
-    auto kernel = kernel_hausdorff<T>;
+    auto d_xs = cudf::column_device_view::create(xs);
+    auto d_ys = cudf::column_device_view::create(ys);
 
-    kernel<<<grid, MAX_NUM_THREADS_PER_BLOCK, 0, stream>>>(points_per_space.size(),
-                                                           xs.data<T>(),
-                                                           ys.data<T>(),
-                                                           space_offsets.data().get(),
-                                                           result->mutable_view().data<T>());
+    auto hausdorff_acc_iter = thrust::make_transform_iterator(
+      hausdorff_index_iter, hausdorff_index_to_acc_functor<T>{*d_xs, *d_ys});
 
-    CUDA_TRY(cudaGetLastError());
+    // ===== Materialize ===========================================================================
+
+    std::unique_ptr<cudf::column> result = make_column<T>(num_results, stream, mr);
+
+    auto result_temp      = rmm::device_buffer(sizeof(hausdorff_acc<T>) * num_results);
+    auto result_temp_iter = static_cast<hausdorff_acc<T>*>(result_temp.data());
+
+    auto scatter_map = thrust::make_transform_iterator(
+      hausdorff_index_iter, [] __device__(hausdorff_index acc) { return acc.result_idx; });
+
+    // the following output iterator and `inclusive_scan_by_key` could be replaced by a
+    // reduce_by_key, if it supported noncommutative operators.
+
+    auto scatter_out = make_scatter_output_iterator(result_temp_iter, scatter_map);
+
+    thrust::inclusive_scan_by_key(rmm::exec_policy(stream)->on(stream),
+                                  hausdorff_acc_iter,
+                                  hausdorff_acc_iter + num_distances,
+                                  hausdorff_acc_iter,
+                                  scatter_out,
+                                  hausdorff_key_compare<T>{});
+
+    thrust::transform(rmm::exec_policy(stream)->on(stream),
```

Comment:
This cast can't be fused with the above scan?

Reply:
https://github.com/thrust/thrust/issues/1199

Haven't tested with latest Thrust yet, but it does not work with `1.9.10`.